From 226270382634ba5712394137aa1d6967dfdd4621 Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Mon, 1 Oct 2012 11:39:12 +0530
Subject: Playtime error after draw the progress bar to end.

BZ: 52514

This patch contains change in AACExtractor.cpp . The
seekTime after drawing the progress bar to end was found
to be more than total duration of file. As a result zero
offset was returned and playback is started from the
start.

Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: Yes

Orig-Change-Id: Ib72b8ad50a5423e83bae21463fa6648d502304b0

Change-Id: I18d81cba569514572a93c7a4680f844fec93e7a5
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Reviewed-on: http://android.intel.com:8080/77933
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libstagefright/AACExtractor.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AACExtractor.cpp b/media/libstagefright/AACExtractor.cpp
index 4d1072f..0fb7bbf 100644
--- a/media/libstagefright/AACExtractor.cpp
+++ b/media/libstagefright/AACExtractor.cpp
@@ -288,8 +288,15 @@ status_t AACSource::read(
     *out = NULL;
 
     int64_t seekTimeUs;
+    int64_t durationUs = 0;
     ReadOptions::SeekMode mode;
     if (options && options->getSeekTo(&seekTimeUs, &mode)) {
+        mMeta->findInt64(kKeyDuration, &durationUs);
+        if (durationUs > 0 && durationUs <= seekTimeUs) {
+            ALOGE("seekTime more than the total duration of File");
+            return ERROR_END_OF_STREAM;
+        }
+
         if (mFrameDurationUs > 0) {
             int64_t seekFrame = seekTimeUs / mFrameDurationUs;
             mCurrentTimeUs = seekFrame * mFrameDurationUs;
-- 
1.7.4.1

